Movecraft/src/main/java/net/countercraft/movecraft/Movecraft.java

```diff
@@ -227,117 +97,33 @@ public void onEnable() {
-            logger.warning("Conflicting datapack already exists in " + datapackDirectory.getPath() + ". If you would like to regenerate the datapack, delete the existing one.");
-            return false;
-        }
-        if(!datapackDirectory.canWrite()) {
-            logger.warning("Missing permissions to write to world directory.");
-            return false;
-        }
-
-        try(var stream = new FileOutputStream(new File(datapackDirectory, "movecraft-data.zip"));
-                var pack = getResource("movecraft-data.zip")) {
-            if(pack == null) {
-                logger.severe("No internal datapack found, report this.");
-                return false;
-            }
-            pack.transferTo(stream);
-        }
-        catch(IOException e) {
-            e.printStackTrace();
-            return false;
-        }
-        logger.info("Saved default Movecraft datapack.");
-
-        getServer().dispatchCommand(getServer().createCommandSender(response -> {}), "datapack list"); // list datapacks to trigger the server to check
-        for (Datapack datapack : getServer().getDatapackManager().getPacks()) {
-            if (!datapack.getName().equals("file/movecraft-data.zip"))
-                continue;
-
-            if (!datapack.isEnabled()) {
-                datapack.setEnabled(true);
-                logger.info("Datapack enabled.");
-            }
-            break;
-        }
-
-        if (!isDatapackEnabled()) {
-            logger.severe("Failed to automatically load movecraft datapack. Check if it exists.");
-            setEnabled(false);
-            return false;
-        }
-        return true;
-    }
-
-    private boolean isDatapackEnabled() {
-        getServer().dispatchCommand(getServer().createCommandSender(response -> {}), "datapack list"); // list datapacks to trigger the server to check
-        for (Datapack datapack : getServer().getDatapackManager().getPacks()) {
-            if (!datapack.getName().equals("file/movecraft-data.zip"))
-                continue;
-
-            return datapack.isEnabled();
-        }
-        return false;
-    }
-
-
-    public WorldHandler getWorldHandler(){
-        return worldHandler;
+    public @NotNull WorldHandler getWorldHandler(){
+        return application.container().getService(WorldHandler.class);
     }
 
-    public SmoothTeleport getSmoothTeleport() {
-        return smoothTeleport;
+    public @NotNull SmoothTeleport getSmoothTeleport() {
+        return application.container().getService(SmoothTeleport.class);
     }
 
-    public AsyncManager getAsyncManager() {
-        return asyncManager;
+    public @NotNull AsyncManager getAsyncManager() {
+        return application.container().getService(AsyncManager.class);
     }
 
     public @NotNull WreckManager getWreckManager(){
-        return wreckManager;
+        return application.container().getService(WreckManager.class);
```

Comment:
Should we deprecate these and instead expose `getService`?

Reply:
Nah but it might be good to expose an API class that holds this stuff or something - exposing the DI container is an anti-pattern typically